From cd762f6b0cee2301f3002541fb28723b6a263b00 Mon Sep 17 00:00:00 2001
From: Wulsic <wulsic@hotmail.com>
Date: Sun, 21 Dec 2014 16:58:15 +0100
Subject: [PATCH 1/1] FFMPEG Backwards compatibility with non neon devices This
 patch will remove/disable neon related code which allows us to compile again
 without errors due to the fact that our device doesn't support neon code.

Change-Id: I482f3792a0987f7e1ec59513ae5d217abaf3d292
---
 android/config-armv7-a.mak |  6 +++---
 android/include/config.h   | 10 +++++-----
 libavcodec/Android.mk      |  2 --
 3 files changed, 8 insertions(+), 10 deletions(-)

diff --git a/android/config-armv7-a.mak b/android/config-armv7-a.mak
index 076e343..32f2a0c 100644
--- a/android/config-armv7-a.mak
+++ b/android/config-armv7-a.mak
@@ -7,7 +7,7 @@ ARCH_ARM:=yes
 HAVE_ARMV5TE:=yes
 HAVE_ARMV6:=yes
 HAVE_ARMV6T2:=yes
-HAVE_NEON:=yes
+HAVE_NEON:=no
 HAVE_VFP:=yes
 HAVE_VFPV3:=yes
 !HAVE_AMD3DNOW:=yes
@@ -27,7 +27,7 @@ HAVE_VFPV3:=yes
 HAVE_ARMV5TE_EXTERNAL:=yes
 HAVE_ARMV6_EXTERNAL:=yes
 HAVE_ARMV6T2_EXTERNAL:=yes
-!HAVE_NEON_EXTERNAL:=yes
+!HAVE_NEON_EXTERNAL:=no
 HAVE_VFP_EXTERNAL:=yes
 HAVE_VFPV3_EXTERNAL:=yes
 !HAVE_AMD3DNOW_EXTERNAL:=yes
@@ -46,7 +46,7 @@ HAVE_VFPV3_EXTERNAL:=yes
 HAVE_ARMV5TE_INLINE:=yes
 HAVE_ARMV6_INLINE:=yes
 HAVE_ARMV6T2_INLINE:=yes
-!HAVE_NEON_INLINE:=yes
+!HAVE_NEON_INLINE:=no
 HAVE_VFP_INLINE:=yes
 HAVE_VFPV3_INLINE:=yes
 !HAVE_AMD3DNOW_INLINE:=yes
diff --git a/android/include/config.h b/android/include/config.h
index 4602acd..7d05b46 100644
--- a/android/include/config.h
+++ b/android/include/config.h
@@ -1,7 +1,7 @@
 /* Automatically generated by configure - do not modify! */
 #ifndef FFMPEG_CONFIG_H
 #define FFMPEG_CONFIG_H
-#define FFMPEG_CONFIGURATION "--arch=arm --target-os=linux --enable-cross-compile --cross-prefix=/a/dev/lollipop/prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.8/bin/arm-linux-androideabi- --sysroot=/a/dev/lollipop/prebuilts/ndk/current/platforms/android-19/arch-arm --enable-shared --enable-static --enable-avresample --extra-cflags='-march=armv7-a -mfloat-abi=softfp -mfpu=neon' --extra-ldflags='-Wl,--fix-cortex-a8'"
+#define FFMPEG_CONFIGURATION "--arch=arm --target-os=linux --enable-cross-compile --cross-prefix=/a/dev/lollipop/prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.8/bin/arm-linux-androideabi- --sysroot=/a/dev/lollipop/prebuilts/ndk/current/platforms/android-19/arch-arm --enable-shared --enable-static --enable-avresample --extra-cflags='-march=armv7-a -mfloat-abi=softfp' --extra-ldflags='-Wl,--fix-cortex-a8'"
 #define FFMPEG_LICENSE "LGPL version 2.1 or later"
 #define CONFIG_THIS_YEAR 2014
 #define FFMPEG_DATADIR "/usr/local/share/ffmpeg"
@@ -54,7 +54,7 @@
 #define HAVE_ARMV6 1
 #define HAVE_ARMV6T2 1
 #define HAVE_ARMV8 0
-#define HAVE_NEON 1
+#define HAVE_NEON 0
 #define HAVE_VFP 1
 #define HAVE_VFPV3 1
 #define HAVE_SETEND 1
@@ -89,7 +89,7 @@
 #define HAVE_ARMV6_EXTERNAL 1
 #define HAVE_ARMV6T2_EXTERNAL 1
 #define HAVE_ARMV8_EXTERNAL 0
-#define HAVE_NEON_EXTERNAL 1
+#define HAVE_NEON_EXTERNAL 0
 #define HAVE_VFP_EXTERNAL 1
 #define HAVE_VFPV3_EXTERNAL 1
 #define HAVE_SETEND_EXTERNAL 1
@@ -124,7 +124,7 @@
 #define HAVE_ARMV6_INLINE 1
 #define HAVE_ARMV6T2_INLINE 1
 #define HAVE_ARMV8_INLINE 0
-#define HAVE_NEON_INLINE 1
+#define HAVE_NEON_INLINE 0
 #define HAVE_VFP_INLINE 1
 #define HAVE_VFPV3_INLINE 1
 #define HAVE_SETEND_INLINE 1
@@ -223,7 +223,7 @@
 #define HAVE_UNISTD_H 1
 #define HAVE_WINDOWS_H 0
 #define HAVE_WINSOCK2_H 0
-#define HAVE_INTRINSICS_NEON 1
+#define HAVE_INTRINSICS_NEON 0
 #define HAVE_ATANF 1
 #define HAVE_ATAN2F 1
 #define HAVE_CBRT 1
diff --git a/libavcodec/Android.mk b/libavcodec/Android.mk
index 83f348c..d02e802 100644
--- a/libavcodec/Android.mk
+++ b/libavcodec/Android.mk
@@ -20,8 +20,6 @@ LOCAL_SHARED_LIBRARIES +=	\
 	libavutil \
 	libswresample
 
-LOCAL_SRC_FILES += neon/mpegvideo.c
-
 # It's strange wmalosslessdec.c can't be compiled by -O3 for armv7-a-neon. A gcc bug?
 $(intermediates)/wmalosslessdec.o: PRIVATE_CFLAGS += $(if $(filter arm,$(TARGET_ARCH)),-Os)
 
-- 
1.9.1

